core/src/edu/cornell/gdiac/main/PlayerModel.java

```diff
@@ -196,6 +207,16 @@ public boolean isFrozen() {
    */
   public void setFrozen(boolean value) {
     // TODO P3 update frozen and apply physics, also visually change player in some way
+    isFrozen = value;
+    if (isFrozen){
+      setDensity(getDensity() * 3);
```

Comment:
After doing a temporary fix to the movement issue I tried freezing the character and it didn't appear to do much other than preventing my character from moving. Maybe try applying a negative vertical force to the player. You can figure out what you want to do to make this happen but just make the effect more noticable.